src/main/java/org/jabref/model/database/shared/DatabaseSynchronizer.java

```diff
@@ -0,0 +1,24 @@
+package org.jabref.model.database.shared;
+
+import java.sql.SQLException;
+
+import org.jabref.model.entry.BibEntry;
+
+public interface DatabaseSynchronizer {
+
+    String getDBName();
+
+    void pullChanges();
+
+    void closeSharedDatabase();
+
+    void registerListener(Object listener);
+
+    void openSharedDatabase(DatabaseConnection connection) throws DatabaseNotSupportedException, SQLException;
```

Comment:
I would let this rather be more general, by not having the concrete SQLException here. I could implement this interface for my Sharelatex Connection.

Reply:
Ok, I've inlined the `SQLException` in favor of an `IllegalStateException`.